src/Twig/Extension/BasicExtension.php

```diff
@@ -34,12 +34,9 @@ class BasicExtension extends AbstractExtension
     public function getFilters(): array
     {
         return [
-            new TwigFilter('debug', 'debug'),
-            new TwigFilter('pr', 'pr'),
             new TwigFilter('low', 'strtolower'),
             new TwigFilter('up', 'strtoupper'),
             new TwigFilter('env', 'env'),
-            new TwigFilter('count', 'count'),
```

Comment:
Won't these removals break applications using these filters? Should we keep them and emit deprecation warnings? Or flag them as deprecated in the docs for a bit before removing them?

Reply:
We can create a deprecation wrapper for these and remove in 2.0 if you want. This follows a previous PR that removed filters that hid existing twig filters.

https://github.com/cakephp/twig-view/pull/56

We can't really deprecate those so should we just release 2.0 with api fixes?

Reply:
A 2.0 triggers a lot of cascading majors in relying libraries and plugins
I prefer a .x and a good changelog so people are aware.

Reply:
Depredations would only be needed for the functions that you are removing. Removing the twig function overloads could change formatting which isn't entirely an API break. What I would like to avoid is fatal errors as people upgrade.

Reply:
The overloads that are removed aren't compatible. I don't know what error would be thrown.

Reply:
markstory isn't concerned about removal of overloaded functions. He's concerned about removal of additional functions like `count()`. Instead of simply removing them they should be deprecated and throw warning.

Reply:
Sure. I'm just saying those will probably throw errors too.

Reply:
Added deprecation warnings.